src/scheduler/tasks/SynchronizeUsersImportTask.ts

```diff
@@ -0,0 +1,106 @@
+import User, { UserImportStatus, UserRole, UserStatus } from '../../types/User';
+
+import Constants from '../../utils/Constants';
+import { LockEntity } from '../../types/Locking';
+import LockingManager from '../../locking/LockingManager';
+import Logging from '../../utils/Logging';
+import SchedulerTask from '../SchedulerTask';
+import { ServerAction } from '../../types/Server';
+import { TaskConfig } from '../../types/TaskConfig';
+import Tenant from '../../types/Tenant';
+import UserStorage from '../../storage/mongodb/UserStorage';
+import UserValidator from '../../server/rest/v1/validator/UserValidation';
+import global from '../../types/GlobalType';
+
+const MODULE_NAME = 'SynchronizeUsersImportTask';
+
+export default class SynchronizeUsersImportTask extends SchedulerTask {
+  async processTenant(tenant: Tenant, config: TaskConfig): Promise<void> {
+    const synchronizeUsersImport = LockingManager.createExclusiveLock(tenant.id, LockEntity.USER, 'synchronize-users-import');
+    if (await LockingManager.acquire(synchronizeUsersImport)) {
+      try {
+        const importedUsers = await UserStorage.getImportedUsers(tenant.id, { statuses: [UserImportStatus.UNKNOWN] }, Constants.DB_PARAMS_MAX_LIMIT);
+        if (importedUsers.count > 0) {
+          for (const importedUser of importedUsers.result) {
+            const foundUser = await UserStorage.getUserByEmail(tenant.id, importedUser.email);
+            if (foundUser) {
+              importedUser.status = UserImportStatus.ERROR;
+              importedUser.error = 'Email already exists';
+              await UserStorage.saveImportedUser(tenant.id, importedUser);
+            } else {
+              const user: Partial<User> = {
+                firstName: importedUser.firstName,
+                name: importedUser.name.toUpperCase(),
```

Comment:
I'v removed it from here. because it's already converted to uppercase in the upload.
As discussed the name of the user is always uppercase in the project.